src/transformStateWithClones.js

```diff
@@ -4,10 +4,36 @@
  * @param {Object} state
  * @param {Object[]} actions
  *
- * @return {Object[]}
+ * @return {Object[]}\
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const newState = Object.assign({}, state);
+
+  for (const action of actions) {
```

Comment:
try to use destructuring  here
`const { type, extraData, keysToRemove } = action;`

use these variables instead